packages/mask/src/plugins/Trader/SNSAdaptor/trader/hooks/useUpdateBalance.ts

```diff
@@ -1,71 +1,52 @@
-import type { ChainId } from '@masknet/web3-shared-evm'
-import { AllProviderTradeActionType, AllProviderTradeContext } from '../../../trader/useAllProviderTradeContext'
 import {
+    ChainId,
+    useChainBalance,
     EthereumTokenType,
     isSameAddress,
     useAccount,
     useProviderType,
     useTokenConstants,
 } from '@masknet/web3-shared-evm'
-import { useAsync } from 'react-use'
-import { currentBalancesSettings } from '../../../../Wallet/settings'
-import Services from '../../../../../extension/service'
-import { WalletRPC } from '../../../../Wallet/messages'
+import { AllProviderTradeActionType, AllProviderTradeContext } from '../../../trader/useAllProviderTradeContext'
+import { useEffect } from 'react'
 
-export function useUpdateBalance(chainId: ChainId, currentChainId: ChainId) {
+export function useUpdateBalance(chainId: ChainId) {
     const currentAccount = useAccount()
     const currentProvider = useProviderType()
     const { NATIVE_TOKEN_ADDRESS } = useTokenConstants()
+    const balance = useChainBalance(currentAccount, chainId, currentProvider)
 
     const {
         tradeState: [{ inputToken, outputToken }, dispatchTradeStore],
     } = AllProviderTradeContext.useContainer()
 
-    return useAsync(async () => {
-        if (!currentAccount) {
-            dispatchTradeStore({
-                type: AllProviderTradeActionType.UPDATE_INPUT_TOKEN_BALANCE,
-                balance: '0',
-            })
-
-            dispatchTradeStore({
-                type: AllProviderTradeActionType.UPDATE_OUTPUT_TOKEN_BALANCE,
-                balance: '0',
-            })
-            return
-        }
-
-        if (chainId && currentProvider && currentAccount) {
-            const cacheBalance = currentBalancesSettings.value[currentProvider]?.[chainId]
```

Comment:
`            const cacheBalance = currentBalancesSettings.value[currentProvider]?.[chainId]`

In `Swapbox` (aka popup pages), we should use `currentMaskWalletBalancesSettings` instead, so refactor with `useChainBalance`

Reply:
`useChainBalance` doesn't meet. We need to change the balance when `currentChainId` be changed.

Reply:
`useChainBalance` can accept a chainId as parameter